### ISSUE TITLE
Explanation of global_emb unclear

### ISSUE THREAD

**Issue by dany-nonstop** [facebookresearch/PyTorch-BigGraph#144]
I'm experimenting with PBG right now, and noticed that global_emb feature might be useful for my use case. However the document is a little murky in details

> When the global_emb configuration option is active, each entity’s embedding will be translated by a vector that is specific to each entity type (and that is learned at the same time as the embeddings).

now the questions are:

- what are the benefits or use cases of global_emb? I tried to look at the code where global_emb is indeed used in `MultiRelationEmbedder.adjust_embs()`. I see some global_embs is initialized as ` nn.Parameter(torch.zeros((dim,)))`, but I don't seem to figure out how it's trained.

- and how to define an entity's type to enjoy the benefits? I traced down to RelationSchema where `lhs` and `rhs` has some help text explaining that they are 

> "The type of entities on the right-hand side of this relation, i.e., its key in the entities dict."

it looks like every entity has an entity type, but how was the entity dict generated? 

could somebody give a little more explanation here? I would enjoy reading your explanation and contributing some addition writeup to the document.

**Comment by lw** [comment]
> I don't seem to figure out how it's trained

Just like everything else: we use it in the forward pass, we compute a gradient for it in the backward pass, and then optimize it.

> how was the entity dict generated?

You provide it in the config.

**Comment by dbanka** [comment]
@lw what is the use of adding global embedding? I have trained a model with 4 entity types but I have set global_emb as False. What extra benefit would I get if I set it to true?

**Comment by lw** [comment]
I must admit I don't exactly remember (it has been two years ago). I think that global embeddings might play a role in helping an operator distinguish between different types of entities more easily. In other words, it decomposes each entity's embedding into the sum of a per-entity embedding and a per-type embedding, so that all "common" parts of all the entities of a given type can be extracted away.

As with everything, the usefulness (or lack of it) of such a feature depends on your specific use case, hence I suggest you set up an evaluation framework and use it to compare the two models on your own data.

**Comment by rkindi** [comment]
Closing issue
